README.md

```diff
@@ -33,7 +33,23 @@ the version of mockgen used to generate your mocks.
 
 ## Running mockgen
 
-`mockgen` has two modes of operation: source and reflect.
+`mockgen` has three modes of operation: archive, source and reflect.
+
+### Archive mode
+
+Archive mode generates mock interfaces from a package archive
+file (.a). It is enabled by using the -archive flag, the import
+path is also needed as a non-flag argument. No other flags are
```

Comment:
Can archive mode optionally take a list of interfaces, just like reflect mode? Mocking extra interfaces may introduce extra dependencies to the mock code.

Reply:
Could be, but it also means this mode will derive from the source mode. We will need to manually match the specified interfaces to the resolved types. IMHO, the generated code will not introduce dependencies that are not already transitively depended on (in theory), so it should be just fine.

Reply:
Saying we have:

```Go
package foo

type SFoo struct {
}

type IBar Interface {
  Method1() string
}

type IBaz interface {
  Method2() SFoo
}
```

When we only mock `IBar`, then the mock won't depend on package `foo`, so it can be compiled into a different package `foomock` and imported by the tests in package `foo`. However, when we mock both `IBar` and `IBaz`, if the mock is still compiled into `foomock`, then the mock would have to import package `foo` because a method in `IBaz` returns `foo.SFoo`. Now the tests in `foo` cannot import `foomock`, because of circular dependencies.

Reply:
Well, in this case, I'll just put the generated mocks in foo, say mock_foo_test.go. 

I'm never a big fan of exporting mocks for unit tests in packages. The obvious reason is that you are actually testing against an imaginary counterpart, a passing test doesn't reflect the actual usage of the package being mocked.

If you need to provide something for external packages to test against, create a fake sharing the same logic.

This also follows  https://github.com/golang/go/wiki/CodeReviewComments#interfaces.